.github/workflows/macos.yaml

```
@@ -0,0 +1,38 @@
+name: Test build on MacOS
+
+on:
+  push:
+    branches: [ "master", "auto" ]
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  build:
+
+    runs-on: macos-latest
+
+    steps:
+    - name: Install prerequisite packages
+      run: |
+        brew install \
+          automake coreutils cppunit gawk \
+          gnu-getopt gnu-sed grep \
+          make openldap cyrus-sasl
+    - name: Checkout sources
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 5
+    - name: Prepare build
+      run: |
+        export PKG_CONFIG_PATH="`brew list openldap | grep ldap.pc | head -1 | xargs dirname`:`brew list cyrus-sasl | grep libsasl2.pc | head -1 | xargs dirname`";
+        export CFLAGS="-Wno-compound-token-split-by-macro ";
+        export CPPFLAGS="`pkg-config --cflags ldap libsasl2`";
+        export LDFLAGS="`pkg-config --libs ldap libsasl2`";
+        ./bootstrap.sh && \
+        mkdir btbuild && \
+        cd btbuild && \
+        ../configure
+    - name: Build It!
+      run: gmake -C btbuild -j $(nproc) all && gmake -C btbuild -j $(nproc) check
```

Make should be capable of building multiple targets. Like so:
```suggestion
      run: gmake -C btbuild -j $(nproc) all check
```

... though, `all` should be redundant as `check` is supposed to do that and more.

---

In practice it isn't; the check target randomly fails at high levels of parallelism, I suspect that's due to us mixing noinst_*_SOURCES and *_SOURCES